libs/date/src/utils.tsx

```diff
@@ -258,23 +200,74 @@ export const timeMask = (value: string | null, twelveHours?: boolean) => {
   return twelveHours ? [...mask, ...twelveHoursAddOn] : mask;
 };
 
-export const determineMonthInput = (lang: "hr" | "en", value: string | null) => {
-  const position = lang === "hr" ? 4 : 0;
+export const getMaskFromFormat = (value: string, format: string, twelveHours?: boolean): (string | RegExp)[] => {
+  const monthPosition = format.indexOf("MM");
+  const dayPosition = format.indexOf("dd");
+  const hourPosition = format.indexOf("HH");
+  const minutePosition = format.indexOf("mm");
+
+  const generatedMask: (string | RegExp)[] = [];
+  format.split("").forEach((char, index) => {
+    if (char === "M") {
+      if (monthPosition !== index) {
+        generatedMask.push(determineMonthInput("en", value, index - 1));
```

Comment:
Why this depends on language?
second M is always depends on the value of first M, regardless of the language.